test/performance/odp_random.c

```diff
@@ -200,6 +237,97 @@ static int test_random_perf(void *ptr)
 	return 0;
 }
 
+static inline void random_data_latency(test_global_t *global, int thread_idx,
+				       uint32_t rounds, uint8_t *data, uint32_t size)
+{
+	uint32_t i;
+	int32_t ret;
+	odp_time_t t1, t2, start;
+	uint64_t nsec;
+	odp_random_kind_t type = global->type;
+	uint64_t delay = options.delay;
+	uint64_t min = UINT64_MAX;
+	uint64_t max = 0;
+	uint64_t sum = 0;
+	uint64_t seed = 0;
+
+	start = odp_time_local();
+
+	for (i = 0; i < rounds; i++) {
+		uint32_t pos = 0;
+
+		if (delay)
+			odp_time_wait_ns(delay);
+
+		if ((int)type == PSEUDO_RANDOM) {
+			t1 = odp_time_local_strict();
+			while (pos < size) {
```

Comment:
These inner blocks seem very similar to the `*_data_loop` functions, reuse those?

Reply:
The difference is time measurement: for latency it is measured for every round, for performance it is measured once for all rounds (to avoid measurement overhead).